### PR TITLE
Prepare for release of Microsoft.Azure.Batch Data Plane 9.0

### PR DESCRIPTION
**Pull request by matthchr**
## Description
Update the Azure Batch C# SDK with support for REST API version 2018-08-01.7.0
REST API Spec PR: https://github.com/Azure/azure-rest-api-specs/pull/3657
Note that this **is** a breaking change.

In addition to various other improvements, this PR fixes: #4602, #4475, #2538

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/_metadata/batch_data-plane.txt
Comment (dsgouda) [inline src/SDKs/_metadata/batch_data-plane.txt]: Is the REST spec merged? We cannot accept this PR until the spec is merged to Azure master

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/_metadata/batch_data-plane.txt
Comment (matthchr) [reply src/SDKs/_metadata/batch_data-plane.txt]: It's already merged I just generated from a private branch from before and haven't updated this file.

I can regenerate it if you want.

Comment (dsgouda) [reply src/SDKs/_metadata/batch_data-plane.txt]: That would be much appreciated. If you can pull down latest changes and run `msbuild build.proj` before regenerating that'd be even better

**Review by xingwu1 — APPROVED** [review]

**Comment by matthchr** [comment]
@dsgouda - I did as you asked and regenerated -- can you merge this?

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please run `msbuild build.proj /t:build /p:Scope=SDKs\Batch\DataPlane\Azure.Batch` and commit the changes generated to .props file [here](https://github.com/Azure/azure-sdk-for-net/blob/689a258123446665f2aad28ce11b04d850dc523a/src/SDKs/Batch/DataPlane/AzSdk.RP.props)
If you do not see any changes, simply replace `_2018-03-01.6.1;` with `_2018-08-01.7.0;` in the file

**Comment by dsgouda** [comment]
Looks great apart from a minor comment

**Review by dsgouda — APPROVED** [review]
LGTM will merge when CIs pass

**Comment by matthchr** [comment]
@dsgouda - CI passed

**Pull request merged**
